docs/ref/csrf.txt

```diff
@@ -462,3 +472,30 @@ A number of settings can be used to control Django's CSRF behavior:
 * :setting:`CSRF_FAILURE_VIEW`
 * :setting:`CSRF_HEADER_NAME`
 * :setting:`CSRF_TRUSTED_ORIGINS`
+
+Frequently Asked Questions
+==========================
+
+Is posting an arbitrary CSRF token pair (cookie and POST data) using Firebug, Chrome dev tools, Burp Suite, etc. a vulnerability?
+---------------------------------------------------------------------------------------------------------------------------------
+
+No, this is by design. The comparison between the cookie token and the POST
+token is a server-side check but it's not against any stored server-side value.
+Some security audits flag this as a problem but remember that an attacker
+cannot steal a user's browser's CSRF cookie. "Stealing" or modifying your
+own token isn't a vulnerability.
+
+Is the fact that Django's CSRF protection isn't linked to a session a problem?
+------------------------------------------------------------------------------
+
+No, this is by design. We may link them in the future just so that we don't
+have to argue with auditors about it.
```

Comment:
As I understand it, the benefit of linking the token to the session is that leaked CSRF tokens are invalidated when the session expires. This won't stop attacks, but it can limit the extend of a successful attack if it goes unnoticed.

The trade-off is that AJAX requests must now use other, often more complicated means to acquire the token, rather than getting it from the cookie, and that sessions must be enabled. In both cases, if a token is leaked, invalidating all sessions will directly or indirectly create a new token for authenticated users (logging in will rotate the token), so the gain for this trade-off is not very significant.